src/main/java/Duke.java

```diff
@@ -1,10 +1,230 @@
+    private static void acknowledgeCommand() {
+        Task addedTask = tasks.get(tasks.size() - 1);
+        customPrint("Got it. I've added this task:\n" + addedTask.toString() + "\nNow you have "
+                + tasks.size() + " tasks in the list.");
+    }
+
+    /**
+     * Prints the welcome message.
+     */
+    private static void printWelcomeMessage() {
         String logo = " ____        _        \n"
                 + "|  _ \\ _   _| | _____ \n"
                 + "| | | | | | | |/ / _ \\\n"
                 + "| |_| | |_| |   <  __/\n"
                 + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+
+        // Prints welcome message
+        String welcomeMessage = logo;
+        welcomeMessage += "Hello! I'm Duke\n";
+        welcomeMessage += "What can I do for you?";
+        customPrint(welcomeMessage);
+    }
+
+    /**
+     * Prints the invalid command message.
+     */
+    private static void printError() {
+        customPrint("Sorry! I did not understand your command.");
+    }
+
+
+    /**
+     * Prints out a statement with borders.
+     *
+     * @param statement Statement to be printed.
+     */
+    public static void customPrint(String statement) {
+        System.out.println("+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++");
+        System.out.println(statement.trim()); // Remove any trailing spaces
+        System.out.println("+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++");
+    }
+
+    /**
+     * Prints out all the Tasks with borders.
+     */
+    public static void listTasks() {
+        String taskString = "";
+        taskString += "Here are the tasks in your list:\n";
+
+        for (int i = 0; i < tasks.size(); i++) {
+            Task currentTask = tasks.get(i);
+            taskString += (i + 1) + "." + currentTask.toString() + "\n";
+        }
+        customPrint(taskString);
+    }
+
+    /**
+     * Marks a task as done and calls customPrint to print the completed message.
+     *
+     * @param description String ID of task to be marked as completed.
+     */
+    public static void markDone(String description) {
+        int taskId = Integer.parseInt(description) - 1; // -1 as array index starts from 0
+
+        // Checks for valid taskID
+        if (taskId < 0 || taskId > tasks.size() || tasks.size() == 0) {
+            customPrint("You have entered an invalid task ID!");
+            return;
+        }
+
+        // Checks if task has been completed
+        Task currentTask = tasks.get(taskId);
+        if (currentTask.isDone()) {
+            customPrint("You have already completed the task [" + currentTask.getDescription() + "]!");
+        } else {
+            currentTask.markAsDone(); // Mark current task as done
+            String textToPrint = "Nice! I've marked this task as done:\n";
+            textToPrint += currentTask.toString();
+            customPrint(textToPrint);
+        }
     }
 }
```

Comment:
Code has been modularized well. Variable and Function names have been named in camelCase. 
Numerous comments are also good. Good job!